oz/__init__.py

```diff
@@ -131,7 +130,18 @@ def get_template_namespace(self):
     def initialize(self, **kwargs):
         self.trigger("initialize", **kwargs)
 
+    def set_secure_cookie(self, name, value, **kwargs):
+        if self.settings["use_secure_cookie"]:
+            if "secure" not in kwargs:
+                kwargs["secure"] = True
+            if "httponly" not in kwargs:
+                kwargs["httponly"] = True
+        super(RequestHandler, self).set_secure_cookie(name, value, **kwargs)
+
     def prepare(self):
+        if self.settings["use_hsts"]:
+            self.set_header("Strict-Transport-Security", "max-age=2592000; includeSubDomains")
```

Comment:
If you redirected here, it would not have been sent over at this point. `prepare` is called before the request handler. We don't know what the first request will contain, but this should be secure by default. Without auto-redirecting and thus ending the request early, it is not secure by default.